XamarinCommunityToolkit/Views/GesturePad/Service/GesturePadService.gtk.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Drawing.Drawing2D;
+using System.Drawing.Imaging;
+using System.IO;
+using System.Linq;
+using Xamarin.Forms.Platform.GTK.Extensions;
+using Color = Xamarin.Forms.Color;
+using Point = Xamarin.Forms.Point;
+using Size = Xamarin.Forms.Size;
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+    internal class GesturePadService : IGesturePadService
+    {
+        public Stream GetImageStream(IList<Point> points,
+            Size imageSize,
+            float lineWidth,
+            Color strokeColor,
+            Color backgroundColor)
+        {
+            if (points == null || !points.Any())
+            {
+                return Stream.Null;
+            }
+
+            var image = MaxResizeImage(GetImageInternal(points, lineWidth, strokeColor, backgroundColor),
+                                       (float)imageSize.Width, (float)imageSize.Height);
+            using (image)
+            {
+                var stream = new MemoryStream();
+                image.Save(stream, ImageFormat.Jpeg);
+                stream.Position = 0;
+                return stream;
+            }
+        }
+
+        static Bitmap GetImageInternal(IList<Point> points,
+            float lineWidth,
+            Color strokeColor,
+            Color backgroundColor)
+        {
+            var minPointX = points.Min(p => p.X);
+            var minPointY = points.Min(p => p.Y);
+            var drawingWidth = points.Max(p => p.X) - minPointX;
+            var drawingHeight = points.Max(p => p.Y) - minPointY;
+            var bm = new Bitmap((int)drawingWidth, (int)drawingHeight);
+            using (var gr = Graphics.FromImage(bm))
+            {
+                var drawingBackgroundColor = XamarinColorToDrawingColor(backgroundColor);
+                gr.Clear(drawingBackgroundColor);
+                using (var pen = new Pen(XamarinColorToDrawingColor(strokeColor), lineWidth))
+                {
+                    var path = new GraphicsPath();
+                    for (var i = 0; i < points.Count() - 1; i++)
+                    {
+                        var p1 = XamarinPointToDrawingPoint(points.ElementAt(i));
+                        var p2 = XamarinPointToDrawingPoint(points.ElementAt(i + 1));
+                        path.AddLine(p1.X - (float)minPointX, p1.Y - (float)minPointY, p2.X - (float)minPointX,
+                                     p2.Y - (float)minPointY);
+                    }
+
+                    gr.DrawPath(pen, path);
+                }
+            }
+
+            return bm;
+        }
+
+
+        static System.Drawing.Color XamarinColorToDrawingColor(Color xamarinColor)
+        {
+            var mediaColor = xamarinColor.ToGtkColor();
+
+            byte GetColorInByte(ushort color)
```

Comment:
Can you move this local function to the end of the method? Also, can this be `static`?